### PR TITLE
chore: autolabel more for `A-*`

### PR DESCRIPTION
**Pull request by weihanglo**
### What does this PR try to resolve?

A follow-up of #11664 for most `A-*` labels. We have a [doc here][hackmd] tracking autolabel stuff.

### How should we test and review this PR?

All `trigger_files` fields are based on human knowledge, some labels are hard to determine. 
The rule to apply a label on a file: *What a file does ⊂ What a label describes*. 
This is a bit conservative. Shall we be more aggressive? So that it will hurt us and make us know how to improve.

[hackmd]: https://hackmd.io/cNsQWpKyRSWHlgVAIYUFdg

Some files are not included in any of label. Most of them are utilities or types definitions. I think that makes sense not including them at this moment.

### CODE REVIEW

**Comment by weihanglo** [comment]
cc @Muscraft

File: triagebot.toml [hunk 1]
```diff
@@ -37,6 +37,158 @@ warn_non_default_branch = true
...
+
+[autolabel."A-sparse-registry"]
+trigger_files = ["src/cargo/sources/registry/http_remote.rs"]
+
+[autolabel."A-testing-cargo-itself"]
+trigger_files = [
+    "src/cargo/core/compiler/timings.js",
+    "src/cargo/core/compiler/timings.rs",
+    "src/cargo/util/cpu.rs",
+]
```
Comment (epage) [inline triagebot.toml:181]: cargo-test-support?

**Comment by epage** [comment]
I feel like this change does a couple of things
- Provides a resource to go from topic -> code
- Helps highlight our current organization for improvement
- Helps highlight our current tags for improvement

File: triagebot.toml [hunk 2]
```diff
@@ -37,6 +37,168 @@ warn_non_default_branch = true
...
+[autolabel."A-source-replacement"]
+trigger_files = ["src/cargo/sources/replaced.rs"]
+
+[autolabel."A-sparse-registry"]
+trigger_files = ["src/cargo/sources/registry/http_remote.rs"]
+
+[autolabel."A-testing-cargo-itself"]
+trigger_files = [
+    "crates/cargo-test-macro/",
+    "crates/cargo-test-support/",
```
Comment (Muscraft) [inline triagebot.toml:183]: Should `benches/*` be in here?

**Review by Muscraft — COMMENTED** [review]
Looks good! 

Should we include the label descriptions above the table? It has helped me classify what should go where a few times.

File: triagebot.toml (hunk 2 above)
Comment (weihanglo) [reply triagebot.toml:188]: Done. Also made `benches/benchsuite/benches/workspace_initialization.rs` into `A-workspaces`

**Comment by weihanglo** [comment]
> Do you think we should include the label descriptions above the table? It has helped me classify what should go where a few times. I am not sure it is needed in this PR just something to think about.

Or maybe we could teach triagebot how to manage labels based on `triagebot.toml`? That would be way better than syncing them by hand 😆.

**Comment by ehuss** [comment]
Thanks! 

@bors r+

**Comment by ehuss** [comment]
CI will be blocked until https://github.com/rust-lang/rust/pull/107688 lands in nightly.

**Comment by weihanglo** [comment]
I was wrong. Not yet landed 😅

**Pull request merged**